pynq/lib/logictools/waveform.py

```diff
@@ -176,24 +173,188 @@ def draw_wavedrom(data):
+    else:
+        return _draw_javascript(data)
+
+
+def _dump_json_data(data):
+    """Convert the data into Json dump.
+
+    Parameters
+    ----------
+    data : dict
+        A dictionary of the Json formatted data.
+
+    Returns
+    -------
+    str
+        A Json dump of the original data.
+
+    """
+    return json.dumps(data)
+
+
+def _draw_javascript(data):
+    """Display the waveform using the Wavedrom Javascript.
+
+    This method requires 2 javascript files to be present. We get the relative
+    paths for the 2 files in order to proceed.
+    Users can call this method directly to draw any wavedrom data.
+
+    Parameters
+    ----------
+    data : str
+        A dump of a Json formatted data.
+
     """
-    wavedrom_js = '/js/WaveDrom.js'
-    wavedromskin_js = '/js/WaveDromSkin.js'
-    if not os.path.isfile(PYNQ_JUPYTER_NOTEBOOKS + wavedrom_js):
-        raise RuntimeError('Cannot locate {}.'.format(wavedrom_js))
-    if not os.path.isfile(PYNQ_JUPYTER_NOTEBOOKS + wavedromskin_js):
-        raise RuntimeError('Cannot locate {}.'.format(wavedromskin_js))
+    wavedrom_js = 'wavedrom.js'
+    wavedromskin_js = 'wavedromskin.js'
+
+    if not (_is_javascript_present(wavedrom_js) and
+            _is_javascript_present(wavedromskin_js)):
+        _copy_javascripts()
+
     current_path = os.getcwd()
     relative_path = os.path.relpath(PYNQ_JUPYTER_NOTEBOOKS, current_path)
 
-    htmldata = '<script type="WaveDrom">' + json.dumps(data) + '</script>'
+    htmldata = '<script type="WaveDrom">' + data + '</script>'
     IPython.core.display.display_html(IPython.core.display.HTML(htmldata))
     jsdata = 'WaveDrom.ProcessAll();'
     IPython.core.display.display_javascript(
         IPython.core.display.Javascript(
             data=jsdata,
-            lib=[relative_path + '/js/WaveDrom.js',
-                 relative_path + '/js/WaveDromSkin.js']))
+            lib=[relative_path + '/js/wavedrom.js',
+                 relative_path + '/js/wavedromskin.js']))
+
+
+def _is_javascript_present(javascript_name):
+    """Check whether the Javascripts are present in the notebook folder.
+
+    Parameters
+    ----------
+    javascript_name : str
+        The name of the JS file.
+
+    Returns
+    -------
+    bool
+        True if the specified javascript is present.
+
+    """
+    file_path = os.path.join(PYNQ_JUPYTER_NOTEBOOKS, 'js', javascript_name)
+    return os.path.isfile(file_path)
+
+
+def _copy_javascripts():
+    """Copy the required javascripts from the pynq package to notebook folder.
+
+    This method is only required when rendering the wavedrom using 
+    javascripts. This is not required for PhantomJS.
+
+    """
+    src_folder = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), 'js')
+    dst_folder = PYNQ_JUPYTER_NOTEBOOKS
+    if os.system('cp -rf ' + src_folder + ' ' + dst_folder):
+        raise RuntimeError('Cannot copy the javascripts.')
```

Comment:
If we are going to copy what's wrong with `shutil.copytree`? Forking a shell seems somewhat like overkill